src/v0/destinations/intercom/transform.js

```diff
@@ -170,11 +174,10 @@ function validateAndBuildResponse(message, payload, category, destination) {
       response.body.JSON = removeUndefinedAndNullValues(validateTrack(message, payload));
       break;
     case EventType.GROUP: {
-      buildCustomAttributes(message, payload);
-      response.body.JSON = removeUndefinedAndNullValues(payload);
+      response.body.JSON = removeUndefinedAndNullValues(buildCustomAttributes(message, payload));
       respList.push(response);
-      const attachUserAndCompanyResponse = attachUserAndCompany(message, destination.Config);
-      if (attachUserAndCompanyResponse) {
+      if (checkIfEmailOrUserIdPresent(message)) {
+        const attachUserAndCompanyResponse = attachUserAndCompany(message, destination.Config);
```

Comment:
Update `attachUserAndCompany` function to remove validations. All the return values of a function should always be of same data type. In the current form of `attachUserAndCompany` it returns boolean in case validation fails, otherwise object which is bad practice.